lib/burp/xml/issue.rb

```diff
@@ -1,127 +1,130 @@
+      # tree.
+      def method_missing(method, *args)
+
+        # We could remove this check and return nil for any non-recognized tag.
+        # The problem would be that it would make tricky to debug problems with
+        # typos. For instance: <>.potr would return nil instead of raising an
+        # exception
+        unless supported_tags.include?(method)
+          super
+          return
+        end
+
+        # First we try the attributes. In Ruby we use snake_case, but in XML
+        # CamelCase is used for some attributes
+        translations_table = {
+          background: 'issueBackground',
+          detail: 'issueDetail',
+          remediation_background: 'remediationBackground',
+          remediation_detail: 'remediationDetail',
+          vulnerability_classifications: 'vulnerabilityClassifications',
+          serial_number: 'serialNumber'
+        }
+
+        method_name = translations_table.fetch(method, method.to_s)
+
+        # no attributes in the <issue> node
+        # return @xml.attributes[method_name].value if @xml.attributes.key?(method_name)
+
+        # Then we try simple children tags: name, type, ...
+        tag = @xml.xpath("./#{method_name}").first
+        if tag && !tag.text.blank?
+          if tags_with_html_content.include?(method)
+            return cleanup_html(tag.text)
+          else
+            return tag.text
+          end
+        end
+
+        if (method_name.include?('request') || method_name.include?('response'))
+          requestresponse_child(method_name)
         else
-          return tag.text
+          # nothing found, the tag is valid but not present in this ReportItem
+          return nil
         end
       end
 
-      if (['request', 'response'].include?(method_name))
-        requestresponse_child(method_name)
-      else
-        # nothing found, the tag is valid but not present in this ReportItem
-        return nil
-      end
-    end
+      private
 
-    private
+      # Some of the values have embedded HTML content that we need to strip
+      def tags_with_html_content
+        [:background, :detail, :remediation_background, :remediation_detail, :references, :vulnerability_classifications]
+      end
 
-    # Some of the values have embedded HTML content that we need to strip
-    def tags_with_html_content
-      [:background, :detail, :remediation_background, :remediation_detail, :references, :vulnerability_classifications]
-    end
+      def requestresponse_child(field)
+        field_name, index = field.split('_')
+        index = index.to_i
 
-    def requestresponse_child(field)
-      return 'n/a' unless @xml.at('requestresponse') && @xml.at("requestresponse/#{field}")
+        return 'n/a' unless @xml.xpath('requestresponse')[index]
 
-      xml_node = @xml.at("requestresponse/#{field}")
-      result = "[unprocessable #{field}]"
+        xml_node = @xml.xpath('requestresponse')[index].at(field_name)
```

Comment:
What about

```ruby
xml_node = @xml.at_xpath("requestresponse[#{index + 1}]").at(field_name)
```

So that we don't need to pull all occurrences and then index every time we encounter this tag?